### ISSUE TITLE
Resolve paths with parentheses-style format in OpenApiUrlTreeNode

### ISSUE THREAD

**Issue by irvinesunday** [microsoft/OpenAPI.NET#1277]
Paths with parentheses-style format are not able to be resolved appropriately into the expected nodes. For example: `/directoryRoles(roleTemplateId='{roleTemplateId}')` would resolve to 1 node instead of 2.
This is because we split paths by `/` when generating the tree node: https://github.com/microsoft/OpenAPI.NET/blob/9d82233182d11d64a12bd5c250b881e6e102f4d0/src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs#L158

**Comment by baywet** [comment]
I'm not sure why you'd expect two nodes here? Is it because it's an alternate key and the primary key would give us /directoryRoles/{id} ?

**Comment by irvinesunday** [comment]
> I'm not sure why you'd expect two nodes here? Is it because it's an alternate key and the primary key would give us /directoryRoles/{id} ?

Yes. The primary key would give us the expected 2 nodes/segments --> **directoryRoles** and **id**
whereas in the alternate key we would only get 1 node/segment --> **directoryRoles(roleTemplateId='{roleTemplateId}')**

**Comment by baywet** [comment]
But this is correct in my opinion. Although semantically from OData it maps to indexing into the collection, it's still at the same path segment level from a pure URI/HTTP perspective. Why would you want it to be considered a subsegment/what is it breaking?  (besides not mapping 1 to 1 with the OData semantics)

**Comment by baywet** [comment]
Closing due to the staleness of the issue and priorities shifting
